vlm/src/test/scala/geotrellis/contrib/vlm/geotiff/GeoTiffRasterSourceMultiThreadingSpec.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm.geotiff
+
+import geotrellis.contrib.vlm.Resource
+
+import geotrellis.proj4.CRS
+import geotrellis.raster._
+import geotrellis.raster.resample._
+import geotrellis.util._
+
+import org.scalatest.AsyncFunSpec
+
+import scala.concurrent.{Future, ExecutionContext}
+import scala.concurrent.duration.Duration
+
+
+class GeoTiffRasterSourceMultiThreadingSpec extends AsyncFunSpec {
+  val url = Resource.path("img/aspect-tiled.tif")
+  val source: GeoTiffRasterSource = new GeoTiffRasterSource(url)
+
+  implicit val ec = ExecutionContext.global
+
+  describe("Thread Safety") {
+    it("GeoTiffRasterSource should be threadsafe") {
```

Comment:
The test coverage is not full, can you add separate tests for 
* `read()`
* `readBounds()`
* `readExtents()`

This will help you to understand do you need these locks around iterators or not.